extensions/amp-playbuzz/0.1/amp-playbuzz.js

```diff
@@ -0,0 +1,263 @@
+    this.displayShareBar_ = e.getAttribute('display-share-buttons') === 'true';
+    this.displayComments_ = e.getAttribute('display-comments') === 'true';
+  }
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return layout === Layout.RESPONSIVE;
+    // return layout === Layout.CONTAINER;
+    // return isLayoutSizeDefined(layout);
+  }
+
+  /** @override */
+  createPlaceholderCallback() {
+    const placeholder = this.win.document.createElement('div');
+    setStyles(placeholder, {
+      'max-height': '300px',
+      'height': '300px',
+    });
+    placeholder.setAttribute('placeholder', '');
+    placeholder.appendChild(this.createPlaybuzzLoader_());
+    return placeholder;
+  }
+
+  /** @override */
+  layoutCallback() {
+    const iframe = this.element.ownerDocument.createElement('iframe');
+    this.iframe_ = iframe;
+    iframe.setAttribute('frameborder', '0');
+    iframe.setAttribute('allowtransparency', 'true');
+    iframe.src = this.generateEmbedSourceUrl_();
+
+    this.listenToPlaybuzzItemMessage_('resize_height',
+      this.setElementSize_.bind(this));
+
+    whenDocumentComplete(this.win.document)
+      .then(this.setElementSize_.bind(this));
+
+    this.applyFillContent(iframe);
+    this.element.appendChild(iframe);
+    setStyles(iframe, {
+      'opacity': 0,
+      'min-height': '300px',
+      'height': '300px',
+    });
+    setStyles(this.element, {'height': '300px', 'min-height': '300px'});
+
+    return this.iframePromise_ = this.loadPromise(iframe).then(() => {
+      this.togglePlaceholder(false);
+      this.getVsync().mutate(() => {
+        setStyles(iframe, {'opacity': 1});
+      });
+      this.getViewport().onChanged(
+        utils.debounce(this.sendScrollDataToItem_.bind(this), 250));
+    });
+  }
+
+  createPlaybuzzLoader_() {
+    const doc = this.element.ownerDocument;
+    const createElement = utils.getElementCreator(doc);
+
+    const loaderImage = createElement('img', 'pb_feed_anim_mask');
+    loaderImage.src = logo;
+
+    const loaderText = createElement('div', 'pb_feed_loading_text');
+    loaderText.textContent = 'Loading...';
+
+    const loadingPlaceholder =
+      createElement('div', 'pb_feed_placeholder_container',
+        createElement('div', 'pb_feed_placeholder_inner',
+          createElement('div', 'pb_feed_placeholder_content', [
+            createElement('div', 'pb_feed_placeholder_preloader', loaderImage),
+            createElement('div', 'pb_ie_fixer'),
+            loaderText,
+          ])));
+
+    return loadingPlaceholder;
+  }
+
+  /**
+   * @param {number} height
+   *
+   * @memberOf AmpPlaybuzz
+   */
+  setElementSize_(height) {
+
+    if (isNaN(height) || height === this.itemHeight_) {
+      return;
+    }
+
+    this.itemHeight_ = height; //Save new height
+
+    if (this.win.document.readyState !== 'complete') {
+      return;
+    }
+
+    const heightInPixels = this.itemHeight_ + 'px';
```

Comment:
can you please elaborate what's the difference between `attemptChangeHeight` to `changeHeight` to using `setStyles` inside a vertical synchronizer ?
when should i use each?
thanks